### PR TITLE
1w painfull coordinates transformation struggling

### PR DESCRIPTION
**Pull request by apocarteres**
Finally, MoveController is able to translate TMX coordinates to World Coordinates

### CODE REVIEW

**Comment by apocarteres** [comment]
fixed #46

File: src/Classes/MoveController.cpp [hunk 1]
```diff
@@ -53,13 +45,22 @@ void MoveController::initSprite(cocos2d::Vec2 center) {
...
-  return point;
+  auto sp = objectGroup->getObject(name);
+  auto point = Vec2(sp["x"].asFloat(), sp["y"].asFloat());
+  assert(point != Vec2::ZERO);
+  auto tileSize = map->getTileSize();
+  auto x0y0 = map->getLayer("water")->getTileAt(Vec2(0, 0))->getPosition();
+  auto translatedPoint = Vec2(point.x, x0y0.y - point.y);
+  auto pointToPixels = CC_POINT_POINTS_TO_PIXELS(translatedPoint);
+  auto gridPoint = Vec2(ceilf(pointToPixels.x / tileSize.width * 2.0f), ceilf(pointToPixels.y / tileSize.height));
+  return map->getLayer("water")->getTileAt(Vec2(gridPoint.x, gridPoint.y))->getPosition();
```
Comment (stavenko) [inline src/Classes/MoveController.cpp]: Could we have here `getTileAt(gridPoint)`

Comment (apocarteres) [inline src/Classes/MoveController.cpp]: you mean split findObjectPosition to findGridPostion and findWorldPosition?

Comment (stavenko) [inline src/Classes/MoveController.cpp]: Github shows comment right after intersting string (63). I suppose, that this line could be shorter. :)

Comment (apocarteres) [inline src/Classes/MoveController.cpp]: i see. yeah probably we can shorter such lines somehow

**Pull request merged**